controllers/snapshot/snapshot_adapter.go

```diff
@@ -566,6 +566,11 @@ func (a *Adapter) createCopyOfExistingEnvironment(existingEnvironment *applicati
 
 	err = a.client.Create(a.context, environment)
 	if err != nil {
+		// We don't want to leave the new DTC on the cluster without the matching environment
+		dtcErr := a.client.Delete(a.context, dtc)
+		if dtcErr != nil {
+			return nil, fmt.Errorf("failed to delete dtc %s: %w", dtc.Name, err)
```

Comment:
This hides real error why env creation failed.

also this creation/deletion should have AuditLog IMHO

Reply:
Yep, all good points. I'm still testing out if this resolves the edge cases in the best way so didn't invest too much into refining it - will do that after confirming.